src/transactions/test/TxResultsTests.cpp

```diff
@@ -421,40 +430,48 @@ TEST_CASE("txresults", "[tx][txresults]")
         {
             SECTION("missing operation")
             {
-                auto tx = a.tx({});
-                tx->addSignature(a);
+                auto tx = a.tx();
+                a.sign(tx);
+
                 for_all_versions(*app, [&] {
                     validateTxResults(tx, *app, {0, txMISSING_OPERATION});
                 });
             }
 
             SECTION("too early")
             {
-                auto tx = a.tx({payment(root, 1)});
-                tx->addSignature(a);
-                tx->getEnvelope().tx.timeBounds.activate().minTime =
-                    getCloseTime() + 1;
+                auto rawTx = a.rawTx({payment(root, 1)});
+                rawTx.timeBounds.activate().minTime = getCloseTime() + 1;
+                auto tx = a.unsignedTx(rawTx);
+                a.sign(tx);
+                a.sign(tx);
+
                 for_all_versions(*app, [&] {
                     validateTxResults(tx, *app, {baseFee, txTOO_EARLY});
                 });
             }
 
             SECTION("too late")
             {
-                auto tx = a.tx({payment(root, 1)});
-                tx->addSignature(a);
-                tx->getEnvelope().tx.timeBounds.activate().maxTime =
-                    getCloseTime() - 1;
+                auto rawTx = a.rawTx({payment(root, 1)});
+                rawTx.timeBounds.activate().maxTime = getCloseTime() - 1;
+                auto tx = a.unsignedTx(rawTx);
+                a.sign(tx);
+                a.sign(tx);
+
                 for_all_versions(*app, [&] {
                     validateTxResults(tx, *app, {baseFee, txTOO_LATE});
                 });
             }
 
             SECTION("insufficent fee")
             {
-                auto tx = a.tx({payment(root, 1)});
-                tx->addSignature(a);
-                tx->getEnvelope().tx.fee--;
+                auto rawTx = a.rawTx({payment(root, 1)});
+                rawTx.fee--;
+                auto tx = a.unsignedTx(rawTx);
+                a.sign(tx);
+                a.sign(tx);
```

Comment:
extra signature (that was a bug before)

Reply:
This is in SECTION("double signed"), testing for which error code we will get.